shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/resources/ManagedBridgeStatus.java

```diff
@@ -22,19 +23,21 @@ public class ManagedBridgeStatus extends CustomResourceStatus {
     public static final String AUTHORISATION_POLICY_AVAILABLE = "AuthorisationPolicyAvailable";
     public static final String NETWORK_RESOURCE_AVAILABLE = "NetworkResourceAvailable";
 
-    private static final HashSet<Condition> CONDITIONS = new HashSet<>() {
-        {
-            add(new Condition(ConditionTypeConstants.READY, ConditionStatus.Unknown));
-            add(new Condition(SECRET_AVAILABLE, ConditionStatus.Unknown));
-            add(new Condition(CONFIG_MAP_AVAILABLE, ConditionStatus.Unknown));
-            add(new Condition(KNATIVE_BROKER_AVAILABLE, ConditionStatus.Unknown));
-            add(new Condition(AUTHORISATION_POLICY_AVAILABLE, ConditionStatus.Unknown));
-            add(new Condition(NETWORK_RESOURCE_AVAILABLE, ConditionStatus.Unknown));
-        }
-    };
+    private static Set<Condition> getInitialConditions() {
```

Comment:
As far as I see the existing code uses a constant while now every instance of `ManagedBridgeStatus` has a different instance of `Set` of conditions. Can you please clarify the reason of this change?

Reply:
The issue I realize while writing testcase is that we only have single Condition Set instance which is bounded to every ManagedBridgeStatus instance so any change in condition of one ManagedBridgeStatus reflects to all other ManagedBridgeStatus within memory.

Reply:
Ok thanks for the clarification, so essentially that Set is going to be modified so it cannot be the same instance 👍